pkg/controller/container-runtime-config/helpers.go

```diff
@@ -211,27 +197,24 @@ func updateStorageConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfigura
 
 // updateCRIOConfig decodes the data rendered from the template, merges the changes in and encodes it
 // back into a TOML format. It returns the bytes of the encoded data
-func updateCRIOConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
+func createCRIOdropinConfig(internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
 	tomlConf := new(tomlConfigCRIO)
-	if _, err := toml.DecodeReader(bytes.NewBuffer(data), tomlConf); err != nil {
-		return nil, fmt.Errorf("error decoding crio config: %v", err)
-	}
 
 	if internal.PidsLimit > 0 {
 		tomlConf.Crio.Runtime.PidsLimit = internal.PidsLimit
+	} else {
```

Comment:
I am not a huge fan of coding in the defaults for these values. What would you think about instead creating three new toml structs, each with just one value in it. Then ignition could drop in a overridden, for example, `/etc/crio/crio.conf.d/01-pids-limit` file that only overrides pids_limit.

I am worried about coding in the default for these because we just recently changed log level default, and these values could be left behind, resulting in an incomplete value change.

wdyt

Reply:
Yup, makes sense. I was initially not liking having to do the same thing basically 3 times, but it is better than relying on defaults that might change. Fixed it up :)